specific/cmo_sale_split_quote2many_order/models/sale.py

```diff
@@ -0,0 +1,272 @@
+import openerp.addons.decimal_precision as dp
+from openerp.exceptions import Warning
+
+
+class sale_order(models.Model):
+    _inherit = 'sale.order'
+
+    order_plan_ids = fields.One2many(
+        'sale.order.customer.plan',
+        'quote_id',
+        string='Order Reference List',
+        copy=True,
+    )
+    use_multi_customer = fields.Boolean(
+        string='Use Multiple Customer',
+        states={'done': [('readonly', True)]},
+        default=False,
+        help="It indicates that the sale order has been sent.",
+    )
+    use_merge = fields.Boolean(
+        string='Merge Order Line',
+        states={'done': [('readonly', True)]},
+        default=False,
+        help="Merge order line to be single line.",
+    )
+    amount_untaxed_order_plan  = fields.Float(
+        string='Amount Untaxed',
+        readonly=True,
+        related='amount_untaxed',
+    )
+    customer_amount = fields.Integer(
+        string='Customer Amount',
+        compute='_compute_customer_amount',
+    )
+    sale_order_mode = fields.Selection(
+        [('change_price', 'As 1 Job (change price)'),
+         ('change_quantity', 'As Units (change quantity)')],
+        string='Sale Order Mode',
+        requied=True,
+        states={'done': [('readonly', True)]},
+        default='change_price',
+    )
+
+    @api.multi
+    @api.depends('customer_amount')
+    @api.onchange('order_plan_ids')
+    def _compute_customer_amount(self):
+        for order in self:
+            self.customer_amount = len(self.order_plan_ids)
+
+    @api.multi
+    def merge_sale_order_line(self, amount_price):
+        self.ensure_one()
+        order_lines = self.order_line
+        description = (self.name or '') + ' ' + \
+                      (self.date_order or '') + ' ' + \
+                      (self.project_related_id.project_number or '') + ' ' + \
+                      (self.project_related_id.name or '') + ' ' + \
+                      (self.event_date_description or '') + ' ' + \
+                      (self.venue_description or '')
+        new_line = order_lines[0].copy({
+            'order_lines_group': 'before',
+            'sale_layout_custom_group_id': None,
+            'section_code': None,
+            'product_id': None,
+            'price_unit': amount_price /
+                          (1.0 - (order_lines[0].discount / 100.0)),
+            'product_uom_qty': 1.0,
+            'name': description,
+        })
+        for line in order_lines:
+            line.unlink()
+        res = new_line.write({'order_id': self.id})
+        self._amount_all()
+        return res
+
+    @api.multi
+    def adjust_price_sale_order_line(self, amount_price):
+        self.ensure_one()
+        order_line = self.order_plan_ids
+        if order_line and self.sale_order_mode:
+            if self.sale_order_mode == 'change_quantity':
+                for line in self.order_line:
+                    new_quantity = amount_price / self.amount_untaxed
```

Comment:
`from openerp.tools.float_utils import float_round as round`
i use this code for replace round as a float round